### PR TITLE
[CPU] Math node migration on nGraph.

### PR DESCRIPTION
**Pull request by nshchego**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: inference-engine/src/mkldnn_plugin/nodes/math.cpp [hunk 1]
```diff
@@ -43,52 +49,128 @@ class MathImpl: public ExtLayerBase {
...
+                    ngraph::op::v3::Asinh::type_info,
+                    ngraph::op::v0::Atan::type_info,
+                    ngraph::op::v0::Ceiling::type_info,
+                    ngraph::op::v0::Cos::type_info,
+                    ngraph::op::v0::Cosh::type_info,
+                    ngraph::op::v0::Floor::type_info,
+                    ngraph::op::v0::HardSigmoid::type_info,
+                    ngraph::op::v0::Log::type_info,
+                    ngraph::op::v0::Negative::type_info,
+//                    ngraph::op::v0::Reciprocal::type_info, // Likely will be implemented in the nGraph
```
[the remark this answers is not in the document]
Comment (nshchego) [reply inference-engine/src/mkldnn_plugin/nodes/math.cpp]: Fixed

File: inference-engine/src/mkldnn_plugin/nodes/math.cpp [hunk 2]
```diff
@@ -43,52 +49,128 @@ class MathImpl: public ExtLayerBase {
...
-            else if (math_func == "Selu") mathFunction = Math::Selu;
-            else if (math_func == "Sign") mathFunction = Math::Sign;
-            else if (math_func == "Sin") mathFunction = Math::Sin;
-            else if (math_func == "Sinh") mathFunction = Math::Sinh;
-            else if (math_func == "SoftPlus") mathFunction = Math::SoftPlus;
-            else if (math_func == "Softsign") mathFunction = Math::Softsign;
-            else if (math_func == "Tan") mathFunction = Math::Tan;
-            else
-                THROW_IE_EXCEPTION << layer->name << " Incorrect Math layer type!";
+            if (op->get_type_info() == ngraph::op::v0::Erf::type_info) {
```
[the remark this answers is not in the document]
Comment (nshchego) [reply inference-engine/src/mkldnn_plugin/nodes/math.cpp]: Fixed

File: inference-engine/src/mkldnn_plugin/nodes/math.cpp [hunk 3]
```diff
@@ -237,38 +244,145 @@ class MathImpl: public ExtLayerBase {
...
-        Selu,
-        Sign,
-        Sin,
-        Sinh,
-        SoftPlus,
-        Softsign,
-        Tan
-    };
+    static std::map<const ngraph::DiscreteTypeInfo, std::function<void(const std::shared_ptr<ngraph::Node>&, MathImpl& node)>> initializers;// = {
+//        {ngraph::op::v0::Erf::type_info, [](const std::shared_ptr<ngraph::Node>& op, MathImpl& node) {
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/nodes/math.cpp:248]: ?

Comment (nshchego) [reply inference-engine/src/mkldnn_plugin/nodes/math.cpp]: Removed

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**